File: DSA/Pranvkumar_590011587/Day34_Question2.py
# 2. Subtract 1 from an Integer Using Bitwise
def subtract_one(n):
    mask = 1
    while not (n & mask):
        n = n ^ mask      # Flip 0 to 1
        mask <<= 1
    n = n ^ mask          # Flip first 1 to 0
    return n

# --- Test Cases ---
if __name__ == "__main__":
    # Subtract One
    print("\nSubtract 1 using Bitwise:")
    print("Input: 5 -> Output:", subtract_one(5))  
    print("Input: 8 -> Output:", subtract_one(8))  

File: DSA/VikrantSinghBanwala_590018082/day34_question2.c
#include<stdio.h>
int subtractOne(int n){
    return n ^ 1;
}
int main(){
    int n = 5;
    printf("Input: n = %d\n", n);
    printf("Output: %d\n", subtractOne(n));
    n = 8;
    printf("Input: n = %d\n", n);
    printf("Output: %d\n", subtractOne(n));
    n = 4;
    printf("Input: n = %d\n", n);
    printf("Output: %d\n", subtractOne(n));
    n = 1;
    printf("Input: n = %d\n", n);
    printf("Output: %d\n", subtractOne(n));
    n = 0;
    printf("Input: n = %d\n", n);
    printf("Output: %d\n", subtractOne(n));
    n = -5;
    printf("Input: n = %d\n", n);
    printf("Output: %d\n", subtractOne(n));
    return 0;
}